modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCachePartitionExchangeManager.java

```diff
@@ -4028,61 +3960,32 @@ void trim() {
      * Represents a cache rebalance order that takes into account both values: rebalance order itself and rebalance mode.
      * It is assumed SYNC caches should be rebalanced in the first place.
      */
-    private static class CacheRebalanceOrder implements Comparable<CacheRebalanceOrder> {
-        /** Cache rebalance order. */
-        private int order;
-
-        /** Cache rebalance mode. */
-        private CacheRebalanceMode mode;
-
-        /**
-         * Creates a new instance of CacheRebalanceOrder.
-         *
-         * @param order Cache rebalance order.
-         * @param mode Cache rebalance mode.
-         */
-        public CacheRebalanceOrder(int order, CacheRebalanceMode mode) {
-            this.order = order;
-            this.mode = mode;
-        }
+    private static class CacheRebalanceOrderComparator implements Comparator<CacheGroupContext>, Serializable {
```

Comment:
Yes, this is a common practice to implement `Serializable` interface by a comparator.
In this particular case, `assignsSet` is always a "local map" and will never be serialized/deserialized.
So, it is not required to implement `Serializable`

Reply:
Fixed.